src/adagrams.js

```diff
@@ -1,15 +1,153 @@
+/*constructing a leeterpool object with letter as key and no. of times it can be drawn as value*/
+const LETTERPOOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
```

Comment:
Overall, I'd prefer the logic related to creating an entire pool of letters to be in its own helper function. Something like:
```js
export const createLetterPool = () =>
  const listOfAllLetters = [];

  for (const [letter, letterFrequency] of Object.entries(LETTERPOOL)) {
    for (let i = 0; i < letterFrequency; i++) {
      listOfAllLetters.push(letter);
    }
  }

  return listOfAllLetters;
}
```

Then the logic in `drawLetters` is only responsible for drawing a hand of ten letters and can invoke `createLetterPool`.

How could we simplify the code for `drawLetters`? What if you generate a random index and use it to access a random letter in your letter pool array and add it to `drawnLetters`. After that you can use a strategy of virtually "dividing" the list of all letters in a letter pool array into a 'used' and 'unused' side. Then you can swap a  value to the used side.

```js
export const drawLetters = () => {
  // invoke createLetterPool
  // get random index
  // get random letter from letter pool and add to drawnLetters list
  // swap random letter from unused side to used side of your letter pool
}
```

This can be done in 4 lines and would be a little easier to understand.